### PR TITLE
Fix vicinity option posting on wrong line

### PR DESCRIPTION
**Pull request by adhikasp**
Fixes https://github.com/AmadeusITGroup/sonar-stash/issues/194

### CODE REVIEW

File: src/test/java/org/sonar/plugins/stash/issue/StashDiffReportTest.java [hunk 1]
```diff
@@ -76,7 +89,7 @@ public void testAddReport() {
 
   @Test
   public void testGetType() {
-    assertEquals(IssueType.CONTEXT, report1.getType("path/to/diff1", 20, StashDiffReport.VICINITY_RANGE_NONE));
+    assertEquals(null, report1.getType("path/to/diff1", 20, StashDiffReport.VICINITY_RANGE_NONE));
```
Comment (adhikasp) [inline src/test/java/org/sonar/plugins/stash/issue/StashDiffReportTest.java]: AFAIU, `vicinity=0` means we only care about issue in changed code. So it should be impossible for `StashDiffReport.getType()` to return `CONTEXT` if vicinity was 0.

Comment (t-8ch) [reply src/test/java/org/sonar/plugins/stash/issue/StashDiffReportTest.java]: True. Can you use assertNull() for the null assertions?

Comment (adhikasp) [reply src/test/java/org/sonar/plugins/stash/issue/StashDiffReportTest.java]: Done 👍

**Comment by adhikasp** [comment]
CI dependency check failing...

```
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:3.1.0:check (default-cli) on project sonar-stash-plugin: 
[ERROR] 
[ERROR] One or more dependencies were identified with vulnerabilities: 
[ERROR] 
[ERROR] sonar-plugin-api-6.0.jar/META-INF/maven/com.google.guava/guava/pom.xml: CVE-2018-10237
```

File: pom.xml [hunk 2]
```diff
@@ -85,7 +85,7 @@
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <version>22.0</version>
+      <version>25.0-jre</version>
```
Comment (t-8ch) [inline pom.xml]: I cherry-picked this to master. (With your name and everything)

File: src/main/java/org/sonar/plugins/stash/issue/StashDiffReport.java [hunk 3]
```diff
@@ -35,27 +35,40 @@ public void add(StashDiffReport report) {
...
+        } else if (inVicinityOfChangedDiff(diff, destination, vicinityRange)) {
+          isInContextDiff = true;
         }
       }
     }
-    return null;
+    return isInContextDiff ? IssueType.CONTEXT : null;
+  }
+
+  private static boolean lineIsChangedDiff(StashDiff diff) {
```
Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/issue/StashDiffReport.java]: Can you move this up to the other two utility functions?

File: src/main/java/org/sonar/plugins/stash/issue/StashDiffReport.java [hunk 4]
```diff
@@ -35,27 +35,40 @@ public void add(StashDiffReport report) {
...
           return IssueType.CONTEXT;
-        } else if (destination == diff.getDestination() || includeVicinityIssuesForDiff(diff,
-            destination, vicinityRange)) {
+        } else if (!lineIsChangedDiff(diff)) {
+          // We only care about changed diff
+          continue;
+        } else if (isChangedDiff(diff, destination)) {
           return diff.getType();
+        } else if (inVicinityOfChangedDiff(diff, destination, vicinityRange)) {
+          isInContextDiff = true;
```
Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/issue/StashDiffReport.java]: It should be possible to just `return IssueType.CONTEXT` here and get rid of the state variable and the ternary operator at the end, no?

**Comment by t-8ch** [comment]
Thanks for this!
I changed to base to point to the new `issue/190` branch where I created an end-to-end testcase (and the matching testing framework, which I wanted to create for some time now).

**Comment by t-8ch** [comment]
Also the dependency-check warning is triggered by our sonar-api dependency and not our own.
I will try to configure dependency-check to ignore `provided` dependencies.

**Comment by adhikasp** [comment]
Welp, seems the git history got messed up because `issues/194` doesn't get rebased to current `master`. Trying to use `issues/194` base branch fails the CI check, so I branch of from `master` again. I doesn't exactly know what to do here 😅 

Here is my list of commit, feel free to cherry pick it or whatever you think is convenient.
- 3acfc121fe738ddec90e38534e3d7e7c7f86e9ef
- a80f11d1167c31da102283ece7dc42994ba84a41
- d8037cc08a8e7c013807508c4a3003a7d9ca46d3

**Comment by adhikasp** [comment]
@t-8ch oh and I think you misnamed issue 194 as 190?

**Comment by t-8ch** [comment]
Yes I did.
I will just cherrypick your stuff and do the minor style cleanups myself.

**Comment by t-8ch** [comment]
This is on `master` now. Thanks again!

**Pull request closed without merging**